core-v1/src/main/scala/data/FactTable.scala

```diff
@@ -14,7 +14,7 @@ import scala.collection.immutable.SortedMap
   *
   * @note some expressions can update the fact table for sub-expressions.
   */
-trait FactTable extends Any {
+trait FactTable extends Any with IterableOnce[Fact] {
```

Comment:
This should really be `Iterable`, however, if I want to keep the `universal trait` aspect of this trait, then I can't extend `Iterable` because it doesn't extend `Any`. However, using this as an `Iterable` is not a very common use case and calling `.iterator.toIterable` isn't that complicated so it's good enough.